src/util/VectorMessageSink.hpp

```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+#include "messages/MessageSink.hpp"
+
+namespace chatterino {
+
+class VectorMessageSink final : public MessageSink
+{
+public:
+    VectorMessageSink(MessageSinkTraits traits = {},
+                      MessageFlags additionalFlags = {});
+    ~VectorMessageSink() override;
+
+    void addMessage(
+        MessagePtr message, MessageContext ctx,
+        std::optional<MessageFlags> overridingFlags = std::nullopt) override;
+    void addOrReplaceTimeout(MessagePtr clearchatMessage, QTime now) override;
+
+    void disableAllMessages() override;
+
+    void applySimilarityFilters(const MessagePtr &message) const override;
+
+    MessagePtr findMessageByID(QStringView id) override;
+
+    MessageSinkTraits sinkTraits() const override;
+
+    const std::vector<MessagePtr> &messages() const;
+    std::vector<MessagePtr> takeMessages() &&;
```

Comment:
`&&` here means the user must call this on an rvalue reference (e.g. by calling `std::move(vecotMessageSink)`) first, and we use it here to ensure a VectorMessageSink is never left with a moved-from `messages_` field - is that correct?

Reply:
Yes, that's the intention.